src/System.ServiceModel.Tests.Common/src/testproperties.targets

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- 
+    CreateTestPropertiesFile:
+      This task generates a C# file to initialize the TestProperties.Properties
+      dictionary with the provided name/value pairs.
+      FileName :          the full name of the file to generate
+      TestPropertyItems : item collection of name/value pairs to generate
+                          name = item.ItemSpec
+                          value = item's metadata for "Value" or "DefaultValue" if "Value" is blank
+  -->
+  <UsingTask TaskName="CreateTestPropertiesFile" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v12.0.dll">
+    <ParameterGroup>
+      <FileName ParameterType="System.String" Required="true" />
+      <TestPropertyItems ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System" />
+      <Reference Include="System.IO" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+        
+        // In some build environments, paths such as $(OutputPath) are
+        // not created prior to starting the build.  So we may find
+        // we need to create the folder for the generated file.
+        string outputFolder = Path.GetDirectoryName(FileName);
+        if (!Directory.Exists(outputFolder))
+        {
+            Log.LogMessage("Creating output folder {0}", outputFolder);
+            Directory.CreateDirectory(outputFolder);
+        }
+
+        StringBuilder sb = new StringBuilder();
+	      sb.AppendLine("using System.Collections.Generic;");
+	      sb.AppendLine("namespace System.ServiceModel.Tests.Common");
+	      sb.AppendLine("{");
+	      sb.AppendLine("    public partial class TestProperties");
+	      sb.AppendLine("    {");
+        
+        // Generate a string for each property name.
+        // Code accessing the properties should use these generated names.
+        foreach (var item in TestPropertyItems)
+        {
+            sb.AppendLine(String.Format("        public static readonly string {0}_PropertyName = \"{0}\";", item.ItemSpec));
```

Comment:
Currently, we control exactly the contents of the ItemSpec value (done later in this same file).  The value needs to be scrubbed because it can come in from arbitrary MSBuild properties.  So I don't think validation is required.